enhancements/cluster-logging/loki-tokenized-auth-enablement.md

```diff
@@ -0,0 +1,343 @@
+
+### Implementation Details/Notes/Constraints [optional]
+
+Based on the reconciled `CredentialRequest` resource the Loki Operator will populated the credential secrets as follows plus additional configuration required by Loki to operate with tokenized authentication.
+
+__Note__: The LokiStack components requiring access to object storage are: `ingester`, `querier`, `index-gateway`, `compactor` and `ruler`.
+
+#### AWS STS Support
+
+For AWS assuming the following credentials secret provided by CCO:
+
+```yaml
+apiVersion: v1
+stringData:
+  credentials: |-
+    [default]
+    role_arn = <role ARN>
+    web_identity_token_file = <path to mounted service account token such /var/run/secrets/openshift/serviceaccount/token>
+kind: Secret
+metadata:
+  name: logging-loki-aws-credentials
+  namespace: openshift-logging
+type: Opaque
+```
+
+each Loki container will gain two new evironment variables:
+
+```yaml
+containers:
+  - args:
+    - -target=ingester
+    - -config.file=/etc/loki/config/config.yaml
+    - -runtime-config.file=/etc/loki/config/runtime-config.yaml
+    - -config.expand-env=true
+    env:
+    - name: AWS_ROLE_ARN
+      value: arn:aws:iam::269733383066:role/openshift-logging-loki-operator-controller-manager
+    - name: AWS_WEB_IDENTITY_TOKEN_FILE
+      value: /var/run/secrets/openshift/serviceaccount/token
+```
+
+in addition the `LokiStack` ConfigMap `logging-loki-config` needs to provide an `s3_storage_config` as below:
+
+```yaml
+common:
+  storage:
+    s3:
+      s3: "s3://<region>/<bucketnames>"
+      s3forcepathstyle: false
+```
+
+__Note:__ The above configuration is a workaround as suggested by the [grafana/loki#4269](grafana-loki-issue-4269) issue.
+
+#### Azure WIF Support
+
+For Azure assuming the following credentials secret provided by CCO:
+
+```yaml
+apiVersion: v1
+stringData:
+  azure_client_id: <client id>
+  azure_federated_token_file: <path to mounted service account token such as /var/run/secrets/openshift/serviceaccount/token>
+  azure_region: <region>
+  azure_subscription_id: <subscription id>
+  azure_tenant_id: <tenant id>
+kind: Secret
+metadata:
+  name: logging-loki-aws-credentials
+  namespace: openshift-logging
+type: Opaque
+```
+
+each Loki container will gain the following evironment variables:
+
+```yaml
+containers:
+  - args:
+    - -target=ingester
+    - -config.file=/etc/loki/config/config.yaml
+    - -runtime-config.file=/etc/loki/config/runtime-config.yaml
+    - -config.expand-env=true
+    env:
+    - name: AZURE_CLIENT_ID
+      value: <client id>
+    - name: AZURE_TENANT_ID
+      value: <tenant id>
+    - name: AZURE_REGION
+      value: <azure region>
+    - name: AZURE_SUBSCRIPTION_ID
+      value: <subscription id>
+    - name: AZURE_FEDERATED_TOKEN_FILE
+      value: /var/run/secrets/openshift/serviceaccount/token
+```
+
+__Note:__ The Azure Region value is populated from the azure object storage secret referenced by the `lokistack.spec.storage.secret.name`.
```

Comment:
I think we can remove this, this seems to be from one of the first iterations of this document, region seems to also be in the secret created by CCO so we can use that.